syntax = "proto3";
package xcp.ether;

message EtherProto {
  string kind = 1;
  uint32 schema_version = 2;
  map<string, bytes> payload = 3;
  map<string, bytes> metadata = 4;
  map<string, bytes> extra_fields = 5;
  repeated Attachment attachments = 6;
}

message Attachment {
  string id = 1;
  string uri = 2;
  string media_type = 3;
  string codec = 4;
  repeated uint32 shape = 5;
  string dtype = 6;
  uint64 size_bytes = 7;
  bytes inline_bytes = 8;
}
